Cache commitments in localStorage once they are retrieved

alxdavids opened the pull request:
For #66:
- Caches commitments (for the active configuration) locally in the browser after they are retrieved from GitHub
- Cache is cleared when configuration is changed (or when manually cleared by user)
- Reorganised files to make it clearer where functions are used in the workflow (e.g. issuance.js for when tokens are received by the client, redemption.js for when client redeems tokens).

gabbifish commented on src/ext/redemption.js:
```diff
@@ -0,0 +1,90 @@
+
+// Generates the HMAC used to bind request data to a particular token redemption.
+//
+// Inputs:
```
Make comments consistent with the @param notation in other comments.

alxdavids commented on src/ext/issuance.js:
```diff
@@ -0,0 +1,190 @@
+        cXhr.send();
+    } else {
+        if (!commitments.G || !commitments.H) {
+            throw new Error("[privacy-pass]: cached commitments are incorrectly specified: " + commitments + ", version: " + version);
```
Good point, I guess my thinking here was that we've stored something in the commitments-segregated part of localStorage but it is malformed which is weird.
As you say though, there is no reason why we couldn't just send an xhr to get new versions, and I think this is worth doing.

alxdavids commented on src/ext/background.js:
```diff
@@ -527,10 +527,17 @@ function resetSpendVars() {
+ */
 function isBypassHeader(header) {
+    let newConfigVal = parseInt(header.value);
     if (header.name.toLowerCase() == CHL_BYPASS_SUPPORT && header.value != "0") {
```
oops...

alxdavids commented on src/ext/issuance.js:
```diff
@@ -0,0 +1,190 @@
+//      Contents [][]byte `json:"contents"`
+// }
+//
+// Note that Go will automatically render and decode []byte as base64 encoded
```
I am unsure myself actually, this comment is copy-pasted from the original file so it could be quite old and I'm not exactly sure of its origin. I think I'll rewrite it in the @param format and then remove this part to save confusion.

alxdavids commented:
@gabbifish Thanks for your comments! I've made it so the cache fails-open and retrieves the commitments via XHR now and I've addressed the rest of the comments. I've also moved some extra stuff to issuance.js and removed some unused global variables.

alxdavids commented:
Rebased on master, there were a few problems with merge conflicts and linting so I've fixed these. Merging now.
